cuda_core/cuda/core/experimental/_utils/cuda_utils.pyx

```diff
@@ -283,3 +286,39 @@ class Transaction:
         """
         # pop_all() empties this stack so no callbacks are triggered on exit.
         self._stack.pop_all()
+
+
+# Track whether we've already warned about fork method
+_fork_warning_emitted = False
+
+
+def _check_multiprocessing_start_method():
+    """Check if multiprocessing start method is 'fork' and warn if so."""
+    global _fork_warning_emitted
+    if _fork_warning_emitted:
+        return
```

Comment:
Instead of caching it, would it be better to always call `multiprocessing.get_start_method()` and check it?  I worry that it is a global state that we don't own and it could be changed at arbitrary point in time by the user or any package.